legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/domain/DomainParseTreeWalker.java

```diff
@@ -332,6 +333,19 @@ private Property visitSimpleProperty(DomainParserGrammar.PropertyContext ctx)
         property.defaultValue = ctx.defaultValue() == null ? null : this.visitDefaultValue(ctx.defaultValue().defaultValueExpression());
         property.sourceInformation = this.walkerSourceInformation.getSourceInformation(ctx);
         property.propertyTypeSourceInformation = this.walkerSourceInformation.getSourceInformation(ctx.propertyReturnType().type());
+        if (ctx.aggregation() != null)
+        {
+            if (ctx.aggregation().aggregationType().AGGREGATION_TYPE_COMPOSITE() != null)
+            {
+                property.aggregation = AggregationKind.COMPOSITE;
+            } else if (ctx.aggregation().aggregationType().AGGREGATION_TYPE_SHARED() != null)
+            {
+                property.aggregation = AggregationKind.SHARED;
+            } else if (ctx.aggregation().aggregationType().AGGREGATION_TYPE_NONE() != null)
+            {
+                property.aggregation = AggregationKind.NONE;
+            }
```

Comment:
I still think you need to throw an exception here if you get a non-null aggregation kind you do not understand. For example, what if someone adds a new aggregation kind in the parser grammar without updating this code? Then any usages of the new aggregation kind will be parsed incorrectly. It's much better to throw an exception to make sure you're not letting any cases fall through the cracks.
```suggestion
            if (ctx.aggregation().aggregationType().AGGREGATION_TYPE_COMPOSITE() != null)
            {
                property.aggregation = AggregationKind.COMPOSITE;
            }
            else if (ctx.aggregation().aggregationType().AGGREGATION_TYPE_SHARED() != null)
            {
                property.aggregation = AggregationKind.SHARED;
            }
            else if (ctx.aggregation().aggregationType().AGGREGATION_TYPE_NONE() != null)
            {
                property.aggregation = AggregationKind.NONE;
            }
            else
            {
                throw new EngineException("Unknown aggregation kind", EngineErrorType.PARSER);
            }
```